File: 2006Scape Server/src/main/java/com/rs2/integrations/discord/DiscordActivity.java
package com.rs2.integrations.discord;

import com.rs2.game.players.PlayerHandler;

public class DiscordActivity {

    public static Boolean playerCount = false;
    private static int count = 50;

    public static void updateActivity() {
        try {
            if (JavaCord.token != null && !JavaCord.token.equals("")) {
                if (count == 0) {
                    if (PlayerHandler.getPlayerCount() != 1) {
                        JavaCord.api.updateActivity(PlayerHandler.getPlayerCount() + " Players Online");
                        System.out.println("Discord Activity Updated");
                        count = 100;
                    } else {
                        JavaCord.api.updateActivity(PlayerHandler.getPlayerCount() + " Player Online");
                        System.out.println("Discord Activity Updated");
                        count = 100;
                    }
                } else {
                    count--;
                }
            }

        } catch (Exception e) {
            System.out.println("Could not set Discord activity: " + e);
            System.out.println("Null checks");
            try {
                System.out.println("JavaCord.api: " + JavaCord.api);
                System.out.println("PlayerHandler.getPlayerCount: " + PlayerHandler.getPlayerCount());
            } catch (Exception e2) {
                System.out.println(e2);
            }
        }
    }
}

File: 2006Scape Server/src/main/java/com/rs2/integrations/discord/DiscordActivity.kt
package com.rs2.integrations.discord

import com.rs2.game.players.PlayerHandler

object DiscordActivity {
    @JvmField
    var playerCount = false
    private var count = 50
    @JvmStatic
    fun updateActivity() {
        try {
            if (JavaCord.token != null && JavaCord.token != "") {
                if (count == 0) {
                    if (PlayerHandler.getPlayerCount() != 1) {
                        JavaCord.api!!.updateActivity(PlayerHandler.getPlayerCount().toString() + " Players Online")
                        println("Discord Activity Updated")
                        count = 100
                    } else {
                        JavaCord.api!!.updateActivity(PlayerHandler.getPlayerCount().toString() + " Player Online")
                        println("Discord Activity Updated")
                        count = 100
                    }
                } else {
                    count--
                }
            }
        } catch (e: Exception) {
            println("Could not set Discord activity: $e")
            println("Null checks")
            try {
                println("JavaCord.api: " + JavaCord.api)
                println("PlayerHandler.getPlayerCount: " + PlayerHandler.getPlayerCount())
            } catch (e2: Exception) {
                println(e2)
            }
        }
    }
}